File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length <= 2) return nums.length;
        int result = 2;
        for (int i = 2; i < nums.length; i++) {
            if (nums[i] != nums[result - 2]) {
                nums[result] = nums[i];
                result++;
            }
        }
        return result;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var result = 2;
        for (i in 2 until nums.size) {
            if (nums[i] != nums[result-2]) {
                nums[result] = nums[i]
                result++
            }
        }
        return result
    }
}